yacht.sh

```diff
@@ -0,0 +1,187 @@
+  if [ -z "$max_jobs" ]; then
+    max_jobs="$n_jobs"
+  fi
+  if [ "$n_jobs" -gt "$max_jobs" ]; then # clip the cpu count, otherwise compilation memory might explode
+    n_jobs="$max_jobs"
+  fi
+
+  builder="make"
+  cmake_builder_flag=""
+  if [ -n "$build_system" ]; then
+    if ! hash "$build_system" 2>/dev/null; then
+      echo "ERROR: the requested build system $build_system was not found. Halting..."
+      exit 1
+    fi
+    if [ "$build_system" == "ninja" ]; then
+      cmake_builder_flag="-GNinja"
+    fi
+    builder="$build_system"
+  fi
+
+  # if we ask for test, we need to build first, then we can run tests
+  run_tests=""
+  if [ "$target_name" == "test" ]; then
+    target_name="build_tests"
+    target_arg=""
+    run_tests="echo && echo '=========== Tests ============' &&  ${builder} run_tests"
+  fi
+  cmake_cov_flag=""
+  if [ "$target_name" == "coverage" ]; then
+    target_name="build_tests all"
+    target_arg=""
+    cmake_cov_flag="-DCOVERAGE_FLAGS=ON"
+    run_tests="echo && echo '========= ${builder} tests =========' &&  ${builder} run_coverage"
+  fi
+
+  echo "============================="
+  echo "building ${build_type} target"
+  echo "============================="
+  echo
+
+  source_dir=$(pwd)
+  pushd "${build_path}" &>/dev/null || exit 1
+  echo "=========== CMake ==========="
+  CC="${c_compiler}" CXX="${cpp_compiler}" cmake "${cmake_builder_flag}" "${target_arg}" "${cmake_cov_flag}" -D CMAKE_BUILD_TYPE="${build_type}" "${source_dir}" &&
+    echo &&
+    echo "=========== ${builder} ============" &&
+    ${builder} -j"${n_jobs}" ${target_name} && eval "${run_tests}"
+  # save the error code if one occured.
+  error_code=$?
+  popd &>/dev/null || exit 1
+
+  # propagate the error out if an error code occured.
+  if [ $error_code -ne 0 ]; then
+    exit $error_code
+  fi
+}
+
+
+
+
+build_type=RelWithDebInfo
+be_verbose="false"
+do_format="false"
+
+while test $# -gt 0; do
+  case "$1" in
+  -h | --help)
+    echo "build.sh - Builds the source with cmake and make."
+    echo " "
+    echo "./build.sh [-h|--help] [-r|--release] [-d|--debug] [-c|--clean] TARGET"
+    echo " "
+    echo "options:"
+    echo "  -h, --help                show brief help"
+    echo "  -r, --release             use the Release build type    (mutually exclusive with -d,--debug,-m, and --minsizerel)"
+    echo "  -d, --debug               use the Debug build type      (mutually exclusive with -r,--release,-m, and --minsizerel)"
+    echo "  -m, --minsizerel          use the MinSizeRel build type (mutually exclusive with -r,--release,-d, and --debug)"
+    echo "  -c, --clean               cleans the build directory before compiling"
+    echo "  TARGET                    specify a compile target"
+    echo "  -v, --verbose             use verbose output"
+    echo " "
+    echo "if neither -r,--release,-d,--debug,-m, or --minsizerel are set, then the default build type used is Relwithdebinfo."
```

Comment:
needs to add the `-f` and `--format` flags. Also `build.sh` needs renaming to `yatch.sh`.